compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -34,4 +40,125 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
   return cloned;
 }
 
+namespace sinf
+{
+
+/**
+ * @note  CircleReshape always has two inputs: tensor and shape.
+ *        The shape input can be CircleConst, CircleOutputDummy, or CircleNode.
+ *        - If the shape input is CircleConst, the shape is inferred from the constant.
+ *        - If the shape input is CircleOutputDummy, the shape is inferred from
+ *          the attribute or the node itself.
+ *        - If the shape input is CircleNode, the shape is not inferred.
+ */
+loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
+{
+  const loco::DataType S32 = loco::DataType::S32;
+
+  // CircleReshape node must have reshape/shape
+  if (node->shape() == nullptr)
+  {
+    INTERNAL_EXN("2nd input shape() should not be nullptr");
+  }
+
+  bool should_infer = true;
+  loco::TensorShape output_shape;
+  {
+    // Check if reshape/shape is CircleConst
+    auto const_input = dynamic_cast<luci::CircleConst *>(node->shape());
+    if (const_input != nullptr)
+    {
+      output_shape.rank(const_input->size<S32>());
+
+      for (uint32_t axis = 0; axis < output_shape.rank(); ++axis)
+      {
+        output_shape.dim(axis) = const_input->at<S32>(axis);
+      }
+    }
+    else
+    {
+      // Check if reshape/shape is CircleOutputDummy
+      auto dummy_input = dynamic_cast<luci::CircleOutputDummy *>(node->shape());
+      if (dummy_input != nullptr)
+      {
+        if (node->newShape()->rank() > 0)
+        {
+          output_shape.rank(node->newShape()->rank());
+
+          for (uint32_t axis = 0; axis < output_shape.rank(); ++axis)
+          {
+            output_shape.dim(axis) = node->newShape()->dim(axis);
+          }
+        }
+        else
+        {
+          output_shape = circle_shape(node);
```

Comment:
Also, to check my understanding, This part corresponds to 'get shape from own shape`.

Reply:
> Also, to check my understanding, This part corresponds to 'get shape from own shape`.

Yes, you're right :)

Reply:
Why `own_shape` is remained?
I thought that this will be removed because we implemented more perfect inference logic..!

Reply:
> Why `own_shape` is remained? I thought that this will be removed because we implemented more perfect inference logic..! (ref : [#13912 (comment)](https://github.com/Samsung/ONE/pull/13912#discussion_r1741498230))

`own_shape` is remained to handle [this recipe](https://github.com/Samsung/ONE/blob/d8fa13140c8fc0413f5dbc7771f87a59d4f6067c/res/TensorFlowLiteRecipes/Reshape_000/test.recipe#L1-L20)(`Reshape_000.recipe`) which has no attribute, no input shape.

I'm not sure why this kind of recipe exists, but I think we can remove that recipe.

Reply:
According to https://github.com/Samsung/ONE/issues/13927#issuecomment-2330871749, we may be able to revise the recipe first...!
@zetwhite How about your think? :)

Reply:
@llFreetimell Thank you a lot for helping us :smile:  

I think @jongwonyang followed what i suggested in here : https://github.com/Samsung/ONE/issues/13927#issuecomment-2330856187

> * regard `Reshape X, attribute X` a valid graph
>   
>   * while importing make `Reshape/shape = DummyCircle`
>     
>     * Since reshape IR allows only 2 inputs, there is no way to avoid creating a `DummyCircle`.
>   * while shape inferencing if `Reshape/shape = DummyCircle`,
>     
>     * First, try to shape inference using `attribute`
>     * Second, try to shape inference using `output_shape`

It was hard to make a policy about "no attribute, no shape input" case. 
So, we try not to fix another code(importer, recipes) and do it best in shape inference logic.

Reply:
> So, we try not to fix another code(importer, recipes) and do it best in shape inference logic 

I don't mean to say it should be done this way. For now, I chose an easy way. 
If you have any other thoughts, please feel free to share it.

Reply:
I understood :D
Then, I agree with your direction, let's go with it!